lib/player.js

```diff
@@ -2961,9 +2970,11 @@ shaka.Player.prototype.onSegmentAppended_ = function() {
  *
  * @param {shaka.extern.Variant} variant
  * @param {boolean=} clearBuffer
+ * @param {number=} safeMargin
  * @private
  */
-shaka.Player.prototype.switch_ = function(variant, clearBuffer = false) {
+shaka.Player.prototype.switch_ = function(
+    variant, clearBuffer = false, safeMargin = 0) {
```

Comment:
This is the callback from AbrManager.  If you want to change this, you'll need to change the definitions in `externs/shaka/abr_manager.js` and in SimpleAbrManager to match the new function signature.

Reply:
In SimpleAbrManager I see that the call to the switch function only passes the variant, without the optional parameters. I kept it this way. Is that okay?
I pushed the other modifications, including the updated abr manager extern.

Reply:
That's fine.